src/SignalR/clients/ts/signalr/tests/Utils.ts

```diff
@@ -13,9 +13,16 @@ export function registerUnhandledRejectionHandler(): void {
     });
 }
 
-export function delay(durationInMilliseconds: number): Promise<void> {
+export function delayUntil(timeoutInMilliseconds: number, condition?: () => boolean): Promise<void> {
     const source = new PromiseSource<void>();
-    setTimeout(() => source.resolve(), durationInMilliseconds);
+    let timeWait: number = 0;
+    const interval = setInterval(() => {
+        timeWait += 10;
+        if ((condition && condition() === true) || timeoutInMilliseconds <= timeWait) {
+            source.resolve();
```

Comment:
Shouldn't this fail if `timeoutInMilliseconds <= timeWait` and there was a condition that never succeeded?

Reply:
Yeah, that makes sense.
Done